[Status indicator] Update Status indicator pattern guidance (#4309)
carbon-design-system/carbon-website: src/pages/patterns/status-indicator-pattern/index.mdx

smoothly or if issues arise, such as when a task is completed or fails after
submitting a form. They provide feedback on the product’s overall state through
system-wide notifications.

Shape indicators, as a secondary set of indicators, can be defined differently
depending on the product. Their use cases include helping users prioritize
tasks, like a red triangle for high-priority tasks, or showing the current phase
of a component's lifecycle, such as activated or deactivated. They don’t always
necessarily indicate urgency.

### Anatomy

<Row>
<Column colLg={8}>

![Anatomy images of the five types of status indicators](./images/status-anatomy.png)

</Column>
</Row>

<Row>
<Column colSm={2} colMd={4} colLg={4}>

#### 1. Icon indicator

A. Text label <br /> B. Icon (shape, color, and symbol) <br />

#### 3. Badge indicator

A. Shape with or without number (color) <br /> B. Icon (outline) <br />

</Column>
<Column colSm={2} colMd={4} colLg={4}>

#### 2. Shape indicator

A. Text label <br /> B. Icon (shape and color) <br />

#### 4. Differential indicator

A. Text label <br /> B. Symbol (optional color)\* <br />

</Column>
</Row>

<Caption>

>   *Differential indicators must have either a ‘+’ or ‘-’ sign, a caret, or an

Reviewer: I think its supposed to be double quotes here not single. 
```suggestion
  *Differential indicators must have either a “+” or “-” sign, a caret, or an
```

<Row>
<Column colLg={12}>

![Overview image of status indicators](./images/status-overview.png)

</Column>
</Row>

### Variants

| Variant                                              | Usage                                                                                                                                                                     | Use cases                                                                                             |
| ---------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------- |
| [Icon indicator](#icon-indicator)                    | Used any time the layout offers ample space and the content requires maximum attention. They require an icon, a shape, a meaningful color and a descriptive inline label. | Most often used in notifications, progress indicators, data tables, task lists, and dashboard widgets |

> | [Shape indicator](#shape-indicator)                  | Useful in smaller spaces or when users need to scan large amount of data.                                                                                                 | Most often used in lists, dashboards, data tables, data visualizations, and network diagrams          |

Reviewer: These comments still look unresolved. @thyhmdo

maximum number, more than 5 or 6 indicators begins to tax a user and makes it
difficult to focus.

### Standardization

All status icons must be approved and published in the
[icon library](https://carbondesignsystem.com/elements/icons/library/) to ensure
consistency across products. This guidance includes only the most common status
indicators, helping maintain a unified user experience and preventing
unnecessary variations in the design system.

> If your product needs an icon that’s not currently included, please reach out to

Reviewer: Can we say open an issue instead here and link them to the correct repo?

oranges. It’s not a part of the IBM brand palette and it’s also not included in
the v2 color release because it’s for very selective use in data visualizations
and certain status indicators. Do not use this palette in any other manner in
your layouts.

<ColorPalette type="status-extended" shouldShowControls={false} />

## Icon indicator

Icon indicators consist of an icon, a shape, a meaningful color, and a
descriptive label. They are highly effective for communication and enhance
scannability, especially in content-heavy layouts. Common uses include
notifications, progress indicators, data tables, task lists, and dashboard
widgets.

<Row>
<Column colLg={8}>

![Carbon notification component](./images/status-icon-indicator.png)

<Caption>
  Notifications are the most prevalent example of this type of status indicator.
</Caption>

</Column>
</Row>

These severity levels are not intended to dictate the final status names in your
product. They are suggested based on common use cases across products to help
you define status names according to your system’s needs.

<StatusIndicatorTable attention="high" />

#### Labeling and type pairing

> Icon indicators, also known as ‘contextual’ status indicators, are linked to

Reviewer: I don't think the single quote is right here for contextual, it should be either: 
- `contextual` (code snippet)
- "contextual" (double quote)

</Column>
</Row>

#### Status color palette

The status palette includes all of the colors that can be used to reflect
status. Typically, red represents danger or error, orange represents a serious
warning, yellow represents a regular warning, green represents normal or
success, and blue represents passive notifications, usually involving additional
information and workflow progress. This status color palette also includes gray
and purple to add more depth to the system. Gray indicates drafts or jobs that
haven’t been started, and purple indicates outliers or undefined statuses.

<ColorPalette twoColumn type="status" />

#### Extended status palettes

> This palette is only for added contrast accessibility when using yellows and

Reviewer: I think we should still say these colors are not part of the _brand_ palette and should not be used in other contexts.

oranges.

<ColorPalette type="status-extended" shouldShowControls={false} />

#### Status type

Status indicators have been paired with specific type sizes that work best in
products. While shape indicators can be paired with a larger size, it’s

> recommended to use 14pt or 12pt in smaller spaces or when used as a secondary

Reviewer 2: Should this be px instead of pt?

set of indicators alongside the icon indicator set.

| Variant         | Icon size | Type size |
| --------------- | --------- | --------- |
| Icon indicator  | 20px      | 16pt      |
|                 | 16px      | 14pt      |
| Shape indicator | 16px      | 14pt      |
|                 | 16px      | 12pt      |

## Icon indicator

Icon indicators consist of an icon, a shape, a meaningful color, and a
descriptive label. They are highly effective for communication and enhance
scannability, especially in content-heavy layouts. Common uses include
notifications, progress indicators, data tables, task lists, and dashboard
widgets.

<Row>
<Column colLg={8}>

![Carbon notification component](./images/status-icon-indicator.png)

<Caption>
  Notifications are the most prevalent example of this type of status indicator.
</Caption>

</Column>
</Row>

### Icon indicator statuses

Suggested icon indicator status names are based on common use cases across
products to help guide you in defining status notifications according to your
system’s needs. These names are not intended to dictate the final status names
in your product.

> <StatusIndicatorTable attention="high" />

Reviewer 3: Since we are showing color tokens now instead of color values, I think we should make them inline code snippets like we do on the style tabs. This should also be done in the Shape indicator statuses table.

![Aligned status indicator icons.](images/status-alignment-do.png)

  </DoDont>
  <DoDont type="dont" caption="Do not push icons out of alignment with label length. In this case, where the status indicators are flush right, the two digit labels are pushing the icons out of alignment.">

![Misaligned status indicator icons](images/status-alignment-dont.png)

  </DoDont>
</DoDontRow>

## Shape indicator

Shape indicators use a shape, color, and descriptive label to convey the status
of a device, feature, or version—without relying on a symbol.

Their use cases include helping users prioritize tasks, like a red triangle for
high-priority tasks, or showing the current phase of a component's lifecycle,
such as activated or deactivated. They don’t always necessarily indicate
urgency. Shape indicators are also used in diagrams and are designed to remain
clear even at small sizes.

<Row>
<Column colLg={8}>

![Example of a contextual shape status indicator](./images/status-shape-indicator.png)

<Caption>Example of the status label directly next to the shape</Caption>

</Column>
</Row>

> ### Shape indicator statuses

Reviewer 3: Use a comma in this table instead of a slash when separating multiple color tokens like you do in the Icon indicator statuses table for consistency.

<Row>
<Column colLg={8}>

![Example of shape indicator paired with a label](./images/status_indicator_10.png)

<Caption>Example shows shape indicator paired with a label</Caption>

</Column>
</Row>

#### Alignment

When stacked vertically, ensure the icons remain left-aligned with the
accompanying text for easier scanning.

<DoDontRow>
<DoDont
type="do"
caption="Do place shape indicators before labels; they can be placed after other text only if there is no character count variation.">

![Do place shape indicators before labels; they can be placed after other text only if there is no character count variation.](./images/status-legend-do.png)

</DoDont>

<DoDont
type="dont"
caption="Do not place shape indicators after the labels to avoid pushing them out of alignment.">

![Do not place shape indicators after the labels to avoid pushing them out of alignment](./images/status-legend-dont.png)

</DoDont>
</DoDontRow>

> #### Best practice

Reviewer 3: ```suggestion
#### Best practices
```

Reviewer 3: I know there is one thing under this section right now, but we usually keep this heading plural and there could be more best practices to add in the future.

</DoDont>
</DoDontRow>

## Badge indicator

Badge indicators let the user know that something is new or updated. A badge
status is displayed over a ghost icon button, usually in the header, to indicate
an active notification and is cleared after the user acknowledges the
notification. Depending on your use case, the icon button can open a new page or
launch a modal, pane, or flyout.

<Row>
<Column colLg={8}>

![Example of badge indicators in a global header](./images/status-badge-indicator.png)

<Caption>Examples of badge indicators used for global notifications</Caption>

</Column>
</Row>

#### Badge with number

A numbered badge is used when a count of new or updated items is available, and
it's important for the user to know the exact number of updates. This badge has
a limit of three digits, with the last character using a plus symbol.

The badge with a number can only be used with the large icon button (48px) as
global actions in the UI shell header.

<Row>
<Column colLg={8}>

![Four icons with badges that included numbers ranging from one to three digits. The last badge has a plus.](./images/status-badge-with-number.png)

> <Caption>Badges with numbers range from one to three digits</Caption>

Reviewer 3: ```suggestion
<Caption>Badges with numbers ranging from one to three digits</Caption>
```

</Column>
</Row>

#### Badge without number

A badge without a number (or dot badge) is used when a new notification is
available, but the exact number of notifications is either unknown or irrelevant
to the user. This dot badge is more subtle than the numbered badge but still
effectively draws the user’s attention to the icon button.

> The badge without number is commonly seen in toolbars' icon buttons.

Reviewer 3: ```suggestion
The badge without a number is commonly seen in toolbars' icon buttons.
```

Reviewer: @thyhmdo this might just be something weird with the preview because I don't see it like this in the markdown but there's something weird happening with the text this section. 
<img width="842" alt="image" src="https://github.com/user-attachments/assets/461c6b46-b733-483c-bd58-5f5bcb018d9e" />

Reviewer 3: @thyhmdo I also still see what Anna is saying above.

## Accessibility

### Elements

#### Relying only on color is insufficient

> Relying solely on color to convey status is insufficient, covered by
> [Use of Color](https://www.w3.org/WAI/WCAG22/Understanding/use-of-color.html),
> especially for users with color vision deficiencies. While color can be useful,

Reviewer: Formatting suggestion

```suggestion
Relying solely on color to convey status is insufficient,
especially for users with color vision deficiencies (refer to [Use of Color](https://www.w3.org/WAI/WCAG22/Understanding/use-of-color.html)). While color can be useful,
```

> it must be paired with text or other visual cues. To meet accessibility
> standards, or to pass
> [Non-text Contrast](https://www.w3.org/WAI/WCAG22/Understanding/non-text-contrast),

Reviewer: ```suggestion
it must be paired with text or other visual cues. To meet accessibility
standards, and to pass
[non-text contrast](https://www.w3.org/WAI/WCAG22/Understanding/non-text-contrast),
```

#### Color

Differential indicators are either displaying a positive or a negative value.

> Color is optional in these situations as long as the value has either a "+" or

Reviewer: The quotes here are coming in correctly here.
<img width="100" alt="image" src="https://github.com/user-attachments/assets/7a20f875-b078-4e7a-a046-94a9a418a767">

## Accessibility

### Elements

#### Relying only on color is insufficient

Relying solely on color to convey status is insufficient, especially for users
with color vision deficiencies (refer to
[Use of Color](https://www.w3.org/WAI/WCAG22/Understanding/use-of-color.html)).
While color can be useful, it must be paired with text or other visual cues. To

> meet accessibility standards, and to pass

Reviewer 3: ```suggestion
meet accessibility standards and to pass
```

[non-text contrast](https://www.w3.org/WAI/WCAG22/Understanding/non-text-contrast),
ensure that there’s at least a 3:1 contrast between colors used for status
indicators, as well as between the indicator and the page background. If the
contrast is sufficient, even in grayscale, users should still be able to
differentiate statuses without relying solely on color.

As a result, status indicators should rely on at least two elements: color,
shape, or symbol. When the icon itself meets accessibility standards, pairing it
with text helps users better understand and scan the content.

#### Using symbols alone is acceptable

> Using symbols alone is acceptable, as long as they meet non-text contrast

Reviewer 3: ```suggestion
Using symbols alone is acceptable as long as they meet non-text contrast
```

requirements. Ensure the symbols have at least 3:1 contrast so all users,
including those with visual impairments, can perceive them clearly.

<Row>
<Column colLg={8}>

![Examples of sufficient status indicators](./images/status-accessibility-examples.png)

<Caption>Example shows sufficient status indicators</Caption>

</Column>
</Row>

### Outlines

Our current icon indicators are static and used either inline or alongside other
components. Since they are not interactive and the symbols have sufficient
contrast against their shape colors, outlines are not necessary.

However, shape indicators rely solely on shapes and colors, which might not
provide enough accessibility for screen readers and individuals with low color
vision. Therefore, using outlines and pairing text with shape indicators is
essential to enhance their accessibility, especially with lighter status colors
such as orange and yellow in Carbon light themes.

<Row>
<Column colLg={8}>

![Outlines in icon and shape indicators](./images/status-outlines.png)

<Caption>

>   Example shows the icon indicators do not need outlines while icon indicators

Reviewer 3: ```suggestion
  Example shows the icon indicators do not need outlines, while icon indicators
```

Reviewer 3: @thyhmdo I actually think this caption means to say the following: 

> "Example shows icon indicators do not need outlines, while shape indicators do due to the lack of symbols."

## Accessibility

> ### Elements

Reviewer 3: > As a result, status indicators should rely on at least two elements: color, shape, or symbol. 

I think this means to say: "As a result, status indicators should rely on at least two of the following elements: color, shape, or symbol."
